samza-core/src/main/java/org/apache/samza/processor/StreamProcessor.java

```diff
@@ -55,6 +57,7 @@
 @InterfaceStability.Evolving
 public class StreamProcessor {
   private static final Logger LOGGER = LoggerFactory.getLogger(StreamProcessor.class);
+  private static final String CONTAINER_THREAD_NAME_FORMAT = "processorId-%s-container-thread";
```

Comment:
Please use "Samza StreamProcessor % Thread-%d" for consistency with rest of the thread names. See https://github.com/apache/samza/commit/1dfc5cecdfa358851f01b4ef105ccb0f62e9891c

Reply:
1. The %d in the end is the thread id, which is useful to know if the thread got recreated by the ES. If you think that's not possible here, its OK to leave it out.
2. Even then, your update does not contain what you've suggested above. Please fix.